samples/boards/96b_argonkey/microphone/README.rst

```diff
@@ -0,0 +1,131 @@
+************
+
+This sample just requires the ArgonKey board. The board can be fed
+in either one of the following two ways:
+
+- mezzanine mode, plugging the ArgonKey to HiKey board thru its 96Board
+  low-speed connector
+- standalone mode, feeding directly 5V on P1 connector
+
+References
+**********
+
+- :ref:`96b_argonkey`
+
+Building and Running
+********************
+
+ .. zephyr-app-commands::
+    :zephyr-app: samples/boards/96b_argonkey/microphone
+    :host-os: unix
+    :board: 96b_argonkey
+    :goals: run
+    :compact:
+
+Sample Output
+=============
+
+The example acquires 3s of audio and prints out the PCM stream on COM port.
+A USB to TTL 1V8 serial cable may be attached to the low speed connector on
+the back of the board on P3.5 (TX) and P3.7 (RX).
+
+As soon as the acquisition starts the green led glows. At the end of the
+acquisition both the green and red led goes on and the PCM stream is sent
+on COM port. When the output is completed the green led goes off and the red
+led stays on. The process can be re-iterated only by pressing the RST button.
+
+- acquisition starts:     GRN on  - RED off
+- acquisition ends:       GRN on  - RED on
+- output ends:            GRN off - RED on
+
+The characteristic of the PCM audio are hardcoded in the example:
+
+- 32KHz of sample rate
+- 16 bits per sample
+- 1 channel (mono), as only 1 microphone is available
+
+3 seconds of acquisition @32Kz of sampling rate corresponds to 96000 sample
+(x 16bits each).
+
+Since the MP34DT05 microphone is PDM, we need to apply an oversampling factor,
+which is 64 in this example, resulting in a 2MHz bit clock. During the PCM
+conversion the OpenPDMFilter library is filtering out and decimating the data
+by the same 64 factor.
+
+At the end of the acquisition the PCM data will be printed on the terminal
+emulator in either binary or ascii format. The output is controlled by
+PCM_OUTPUT_IN_ASCII definition, which is off by default.
+
+Binary PCM Output
+-----------------
+
+The ttyUSB0 port must be configured in raw mode in order to avoid to have
+characters 'cooked' out.
+
+ .. code-block:: console
+
+    stty -F /dev/ttyUSB0 115200 raw
+    cat /dev/ttyUSB0 > /tmp/sound.raw
+
+NOTE!
+In case the character 0x0a is interpreted as NL and 0x0d (CR) is added
+then it may be necessary to remove it:
+
+ .. code-block:: console
+
+    dos2unix -f /tmp/sound.raw
+
+
+ASCII PCM Output
+----------------
+
+It is also possible to recompile and to have PCM output in ASCII, which needs
+to be converted in binary later on. The output format is the following:
+
+.. code-block:: console
+
+    ArgonKey test!!
+    -- start
+    0xfbe0,
+    0xfbf0,
+    0xfc0c,
+    0xfc24,
+    0xfc3c,
+    0xfc4c,
+    0xfc68,
+    0xfc48,
+
+    [...]
+
+    0xfb98,
+    0xfb98,
+    0xfbb8,
+    0xfbac,
+    0xfbc4,
+    0xfbe8,
+    0xfbf4,
+    -- end
+
+Play PCM Audio
+--------------
+
+At this point in either case we ended up with a binary PCM file, say sound.raw.
```

Comment:
Change this to:

    Now that we have a binary PCM file (say sound.raw), you can use,
    for example, the audicity open source editor/player to load and play it.